hermes/probe/create/create_file.go

```diff
@@ -0,0 +1,160 @@
+// TODO (#70) add license header and author line
+// TODO (#68) add doc strings
+// TODO (#72) change error types and error messages
+
+package create
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+
+	m "github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	journalpb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+const (
+	begin            = 1    // ID of the first HermesFile
+	end              = 50   // ID of the last HermesFile
+	maxFileSizeBytes = 1000 // maximum allowed file size in bytes
+)
+
+type RandomFile struct {
+	ID   int64 // ID is a positive integer in the range [1,50]
+	Size int   // File size in bytes
+}
+
+type randomFileReader struct {
+	size int // size in bytes
+	i    int // currently reading this byte
+	rand *rand.Rand
+}
+
+func (r *randomFileReader) readDone() bool {
+	return r.i >= r.size
+}
+
+func (r *randomFileReader) bytesLeft() int {
+	return (r.size - r.i)
+}
+
+func (r *randomFileReader) bufferTooLong(buf []byte) bool {
+	return len(buf) > r.bytesLeft()
+}
+
+func (r *randomFileReader) Read(buf []byte) (n int, err error) {
+	if r.readDone() {
+		return 0, io.EOF
+	}
+	b := buf
+	if r.bufferTooLong(b) {
+		b = buf[:r.bytesLeft()]
+	}
+	n, err = r.rand.Read(b) // n is the length  of the buffer
+	if err != nil {
+		return n, err // in this case n=0
+	}
+	r.i += n
+	return n, err
+}
+
+// Warning: NewReader is not thread safe
+func (f *RandomFile) NewReader() *randomFileReader {
+	// ID will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
+	return &randomFileReader{
+		size: f.Size,
+		rand: rand.New(rand.NewSource(f.ID)),
+	}
+}
+
+func NewRandomFile(fileID int64, fileSize int) (*RandomFile, error) {
+	if fileID < begin || fileID > end {
+		return &RandomFile{}, fmt.Errorf("the file ID provided %v wasn't in the required range [%v, %v]", fileID, begin, end)
+	}
+	if fileSize > maxFileSizeBytes || fileSize <= 0 {
+		return &RandomFile{}, fmt.Errorf("invalid argument: RandomFile.Size = %d; want 0 < size <= %d", fileSize, maxFileSizeBytes)
+	}
+	return &RandomFile{ID: int64(fileID), Size: fileSize}, nil
```

Comment:
Unfortunately not because rand.NewSource() requires an int64 as an argument added the casting to int64 there and added this comment. The argument to CreateFile is an int32 because the State Journal contains a map [int32] string.